### PR TITLE
FEATURE: Use new topic bulk actions dropdown on search page

### PR DESCRIPTION
**Pull request by martin-brennan**
We want to get rid of the old topic bulk actions modal
and use the new dropdown (currently gated behind
experimental_topic_bulk_actions_enabled_groups). To do
this we need to use the new dropdown in all places in the
UI.

This commit changes the full page search UI to use the new
topic bulk actions dropdown if experimental_topic_bulk_actions_enabled_groups
is enabled, and makes some minor refactors to make this work.
Also add a spec for both the old and new functionality.

image

### CODE REVIEW

File: app/assets/javascripts/discourse/app/templates/full-page-search.hbs [hunk 1]
```diff
@@ -109,13 +109,22 @@
...
-                @icon="wrench"
-                class="btn-default bulk-select-btn"
-              />
+            {{#if this.bulkSelectHelper.selected}}
+              {{#if this.useNewBulkActions}}
+                <TopicList::TopicBulkSelectDropdown
+                  @bulkSelectHelper={{this.bulkSelectHelper}}
+                  @afterBulkActionComplete={{action
+                    this.afterBulkActionComplete
+                  }}
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/templates/full-page-search.hbs:118]: no need for the `action` helper
```suggestion
                  @afterBulkActionComplete={{
                    this.afterBulkActionComplete
                  }}
```

File: app/assets/javascripts/discourse/app/controllers/full-page-search.js [hunk 2]
```diff
@@ -229,6 +231,11 @@ export default Controller.extend({
     }
   },
 
+  @discourseComputed("currentUser.use_experimental_topic_bulk_actions")
+  useNewBulkActions() {
+    return this.currentUser?.use_experimental_topic_bulk_actions;
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/controllers/full-page-search.js:236]: looks like this controller is missing an explicit `currentUser` service injection - could you add it?

Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/controllers/full-page-search.js:236]: Sure, will add :+1:

File: app/assets/javascripts/discourse/app/templates/full-page-search.hbs (hunk 1 above)
Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/templates/full-page-search.hbs:118]: Man this still trips me up so much, when to use vs. when not. I know in new gjs components we dont have to.

Comment (CvX) [reply app/assets/javascripts/discourse/app/templates/full-page-search.hbs:118]: The only (remaining) case where you need it is when the action doesn't have `@action` decorator and is inside an `actions: {}` object.

**Pull request merged**